calliope/core/preprocess/model_data.py

```diff
@@ -426,6 +428,33 @@ def tech_specific_to_dataset(model_run):
     return data_dict
 
 
+def group_constraints_to_dataset(model_run):
+    data_dict = {}
+
+    # Get a DataFrame with constraint names as index, constraint groups as columns
+    constraint_df = (
+        pd.DataFrame(model_run['group_constraints'])
+          .drop(['techs', 'locs'], axis=0, errors='ignore')
+    )
+
+    for constr_name, constraints in constraint_df.iterrows():
+        constraints = constraints.dropna()  # Ignore where constraint not defined
+
+        if constr_name in checks.defaults.allowed_group_constraints.per_carrier:
+            dims = ['constraint_groups', 'carriers']
+            data = [list(constraints.loc[i].values()) for i in constraints.index]
+        elif constr_name in checks.defaults.allowed_group_constraints.per_cost:
+            dims = ['constraint_groups', 'costs']
+            data = [list(constraints.loc[i].values()) for i in constraints.index]
+        elif constr_name in checks.defaults.allowed_group_constraints.general:
+            dims = ['constraint_groups']
+            data = constraints.values
+
+        data_dict['group_' + constr_name] = {'dims': dims, 'data': data}
```

Comment:
I just ran this with unknown constraint. In that case neither 'dims' nor 'data' are defined in this line, leading to 'UnboundLocalError: local variable 'dims' referenced before assignment'. Should be fixed.

Reply:
Done in c6e0caa